drivers/sensor/max31875/max31875.c

```diff
@@ -0,0 +1,272 @@
+		/* conversion rate */
+		switch (cr) {
+		case 250:
+			value = MAX31875_CONV_PER_SEC_0_25 << MAX31875_CONV_PER_SEC_SHIFT;
+			break;
+
+		case 1000:
+			value = MAX31875_CONV_PER_SEC_1 << MAX31875_CONV_PER_SEC_SHIFT;
+			break;
+
+		case 4000:
+			value = MAX31875_CONV_PER_SEC_4 << MAX31875_CONV_PER_SEC_SHIFT;
+			break;
+
+		case 8000:
+			value = MAX31875_CONV_PER_SEC_8 << MAX31875_CONV_PER_SEC_SHIFT;
+			break;
+
+		default:
+			return -ENOTSUP;
+		}
+
+		ret = max31875_update_config(dev, MAX31875_CONV_PER_SEC_MASK, value);
+		if (ret < 0) {
+			LOG_ERR("Failed to set attribute!");
+			return ret;
+		}
+
+		break;
+
+	default:
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static int max31875_sample_fetch(const struct device *dev,
+			       enum sensor_channel chan)
+{
+	struct max31875_data *data = dev->data;
+	const struct max31875_config *cfg = dev->config;
+	uint16_t val;
+
+	if (chan != SENSOR_CHAN_ALL && chan != SENSOR_CHAN_AMBIENT_TEMP) {
+		LOG_ERR("Invalid channel provided");
+		return -ENOTSUP;
+	}
+
+	if (max31875_reg_read(cfg, MAX31875_REG_TEMPERATURE, &val) < 0) {
+		return -EIO;
+	}
+
+	if (data->config_reg & BIT(MAX31875_DATA_FORMAT_SHIFT)) {
+		data->sample = arithmetic_shift_right((int16_t)val,
+							  MAX31875_DATA_FORMAT_EXTENDED_SHIFT);
+	} else {
+		data->sample = arithmetic_shift_right((int16_t)val,
+							  MAX31875_DATA_FORMAT_NORMAL_SHIFT);
+	}
+
+	return 0;
+}
+
+static int max31875_channel_get(const struct device *dev,
+			      enum sensor_channel chan,
+			      struct sensor_value *val)
+{
+	struct max31875_data *data = dev->data;
+	int32_t uval;
+
+	if (chan != SENSOR_CHAN_AMBIENT_TEMP) {
+		return -ENOTSUP;
+	}
+
+	uval = data->sample * MAX31875_TEMP_SCALE;
+	val->val1 = uval / 1000000;
+	val->val2 = uval % 1000000;
+
+	return 0;
+}
+
+static const struct sensor_driver_api max31875_driver_api = {
+	.attr_set = max31875_attr_set,
+	.sample_fetch = max31875_sample_fetch,
+	.channel_get = max31875_channel_get,
+};
+
+int max31875_init(const struct device *dev)
+{
+	const struct max31875_config *cfg = dev->config;
+	struct max31875_data *data = dev->data;
+
+	if (!device_is_ready(cfg->bus.bus)) {
+		LOG_ERR("I2C dev %s not ready", cfg->bus.bus->name);
+		return -ENODEV;
+	}
+
+	data->config_reg = MAX31875_CONFIG(cfg->data_format, cfg->resolution,
+		cfg->conversions_per_second);
+
+	return max31875_update_config(dev, 0, 0);
+}
+
+
+#define MAX31875_INST(inst) \
+	static struct max31875_data max31875_data_##inst; \
+	static const struct max31875_config max31875_config_##inst = { \
+		.bus = I2C_DT_SPEC_INST_GET(inst), \
+		.conversions_per_second = DT_ENUM_IDX(DT_DRV_INST(inst), conversions_per_second), \
+		.resolution = DT_INST_ENUM_IDX(inst, resolution), \
+		.data_format = DT_INST_PROP(inst, extended_mode), \
+	};\
+	DEVICE_DT_INST_DEFINE(inst, max31875_init, NULL, &max31875_data_##inst, \
+				&max31875_config_##inst, POST_KERNEL, \
+				CONFIG_SENSOR_INIT_PRIORITY, &max31875_driver_api);
```

Comment:
by convention: push all `\` to 80 cols (or align them to the same column at least)